applications/ShapeOptimizationApplication/custom_utilities/input_output/universal_file_io.h

```diff
@@ -77,11 +75,15 @@ class UniversalFileIO
     ///@{
 
     /// Default constructor.
-    UniversalFileIO( ModelPart& designSurface, Parameters& optimizationSettings )
-        : mrDesignSurface( designSurface ),
-          mrOptimizationSettings( optimizationSettings )
+    UniversalFileIO( ModelPart& OutputModelPart, std::string OutputFilenameWithoutExtension, std::string WriteConditionsFlag, Parameters NodalResults )
+        : mrOutputModelPart( OutputModelPart ),
+          mOutputFilenameWithExtension( OutputFilenameWithoutExtension + ".unv" ),
+          mrNodalResults( NodalResults )
     {
-        mOutputFilename = createCompleteOutputFilenameWithPath( optimizationSettings );
+        if(WriteConditionsFlag.compare("WriteElementsOnly")==0 || WriteConditionsFlag.compare("WriteConditionsOnly")==0 )
+            mWriteConditionsFlag = WriteConditionsFlag;        
+        else
+            KRATOS_THROW_ERROR(std::runtime_error,"Wrong value specified for \"WriteConditionsFlag\" in UniversalFileIO. Options are: \"WriteElementsOnly\" or \"WriteConditionsOnly\"","" )
```

Comment:
outdated, better use the new stuff:
`KRATOS_ERROR << "My Message" << std::endl`